### PR TITLE
Merge main into preview 3

### PR DESCRIPTION
**Pull request by hmstepanek**
# Overview
Merge changes from main into preview branch and instrument Lantern vectorstore.

### CODE REVIEW

File: newrelic/config.py [hunk 1]
```diff
@@ -43,7 +42,7 @@
 import newrelic.core.agent
 import newrelic.core.config
 from newrelic.common.log_file import initialize_logging
-from newrelic.common.object_names import expand_builtin_exception_name
+from newrelic.common.object_names import callable_name, expand_builtin_exception_name
```
Comment (hmstepanek) [inline newrelic/config.py:45]:
```suggestion
from newrelic.common.object_names import expand_builtin_exception_name
```

File: newrelic/hooks/external_botocore.py [hunk 2]
```diff
@@ -144,7 +144,7 @@ def create_chat_completion_message_event(
             "response.model": request_model,
             "vendor": "bedrock",
             "ingest_source": "Python",
-            "is_response": True
+            "is_response": True,
```
Comment (hmstepanek) [inline newrelic/hooks/external_botocore.py:147]:
```suggestion
            "is_response": True
```

File: newrelic/hooks/external_botocore.py [hunk 3]
```diff
@@ -243,7 +246,7 @@ def extract_bedrock_claude_model(request_body, response_body=None):
     chat_completion_summary_dict = {
         "request.max_tokens": request_body.get("max_tokens_to_sample", ""),
         "request.temperature": request_body.get("temperature", ""),
-        "response.number_of_messages": len(input_message_list)
+        "response.number_of_messages": len(input_message_list),
```
Comment (hmstepanek) [inline newrelic/hooks/external_botocore.py:249]:
```suggestion
        "response.number_of_messages": len(input_message_list)
```

File: newrelic/hooks/external_botocore.py [hunk 4]
```diff
@@ -271,11 +274,13 @@ def extract_bedrock_cohere_model(request_body, response_body=None):
     chat_completion_summary_dict = {
         "request.max_tokens": request_body.get("max_tokens", ""),
         "request.temperature": request_body.get("temperature", ""),
-        "response.number_of_messages": len(input_message_list)
+        "response.number_of_messages": len(input_message_list),
```
Comment (hmstepanek) [inline newrelic/hooks/external_botocore.py:277]:
```suggestion
        "response.number_of_messages": len(input_message_list),
```
```suggestion
        "response.number_of_messages": len(input_message_list)
```

File: newrelic/hooks/external_botocore.py [hunk 5]
```diff
@@ -363,7 +368,7 @@ def wrap_bedrock_runtime_invoke_model(wrapped, instance, args, kwargs):
                 notice_error_attributes = {
                     "http.statusCode": error_attributes["http.statusCode"],
                     "error.message": error_attributes["error.message"],
-                    "error.code": error_attributes["error.code"]
+                    "error.code": error_attributes["error.code"],
```
Comment (hmstepanek) [inline newrelic/hooks/external_botocore.py:371]:
```suggestion
                    "error.code": error_attributes["error.code"]
```

**Review by umaannamalai — APPROVED** [review]

**Comment by umaannamalai** [comment]
Thanks for adding the Lantern and OpenAI changes to this branch!

**Pull request merged**